apps/emqx_bridge_http/src/emqx_bridge_http_connector.erl

```diff
@@ -1080,7 +1092,10 @@ join_paths_test_() ->
         ?_assert(iolists_equal("/cde", join_paths("/", "cde"))),
         ?_assert(iolists_equal("/cde", join_paths("/", "/cde"))),
         ?_assert(iolists_equal("//cde/", join_paths("/", "//cde/"))),
-        ?_assert(iolists_equal("abc///cde/", join_paths("abc//", "//cde/")))
+        ?_assert(iolists_equal("abc///cde/", join_paths("abc//", "//cde/"))),
+        ?_assert(iolists_equal("abc?v=1", join_paths("abc", "?v=1"))),
+        ?_assert(iolists_equal("abc?v=1", join_paths("abc/", "?v=1"))),
```

Comment:
Since some HTTP servers might treat the presence of a slash differently, I think this should be the case that should pass:

```suggestion
        ?_assert(iolists_equal("abc/?v=1", join_paths("abc/", "?v=1"))),
```

... so maybe if the path starts with a question mark, we should leave the trailing slash?

Reply:
`/?` is uncommon, and I believe in most of the cases, it's a copy-paste left-over.

`without_traling_slash` only removes the last `/`, but not all
so if `/` is indeed intended, one can always add `//` to workaround.

Reply:
fun reading here: https://stackoverflow.com/questions/1617058/ok-to-skip-slash-before-query-string
the most popular answer is in favor of omitting `/` before `?` i.e. it's equivalent with or without the `/`.

Reply:
My reading of that SO discussion is that it's about _empty HTTP URI path_ case, which is slightly more specific than _last slash in HTTP URI path_ case.

That said, I can imagine that some generic HTTP server with near-default config could serve `https://server/abc?v=1` with `/www/root/abc.html` and `https://server/abc/?v=1` with `/www/root/abc/index.html`.

Reply:
@keynslug which server is that ?

Reply:
Hypothetical. 🫠 However, briefly looking into the [nginx documentation](https://nginx.org/en/docs/http/ngx_http_core_module.html#try_files) for example we may find examples like these:
```
location / {
    try_files $uri $uri/index.html $uri.html =404;
}
```
^ IIUC this actually makes trailing slash non-important, yet it's easy to imagine a configuration where the opposite is true.